pkg/cmd/roachtest/cluster/dynamic_cluster_interface.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cluster
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+)
+
+// DynamicCluster is a cluster that can be resized. This is only for use by
+// operations that mutate the underlying cluster topology.
+type DynamicCluster interface {
```

Comment:
Do we need an interface for this? Doesn't seem to be used.

Reply:
This is used in the resize operation, to check if the provided Cluster interface also supports the DynamicCluster interface.